### PR TITLE
statistics/handle: fix trigger auto analyze for partition table after adding index

### LINKED ISSUE #41638: auto analyze of partition table is not triggered after adding index
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
use test;
create table t (a int, b int) partition by range (a) (PARTITION p0 VALUES LESS THAN (2000), PARTITION p1 VALUES LESS THAN MAXVALUE);
-- insert more than 1000 rows into each partition
analyze table t;
alter table t add index idx(a);
show analyze status; -- observe whether auto analyze is triggered after adding the index
```

### 2. What did you expect to see? (Required)
Auto analyze is triggered after adding the index.

### 3. What did you see instead (Required)
Auto analyze is not triggered after adding the index.

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What problem does this PR solve?

Issue Number: close #41638

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

Test #41638 manually.

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/update.go [hunk 1]
```diff
@@ -1230,14 +1230,11 @@ func (h *Handle) autoAnalyzePartitionTableInDynamicMode(tblInfo *model.TableInfo
 		sqlBuilder.WriteString(suffix)
 		return sqlBuilder.String()
 	}
-	if len(partitionNames) < 1 {
```
Comment (qw4990) [inline statistics/handle/update.go:1233]: Why do we remove this?

Comment (xuyifangreeneyes) [reply statistics/handle/update.go:1233]: When the if branch exists, we would not go to the part of checking index stats(old Line1258-1289).

**Review by qw4990 — APPROVED** [review]

**Review by chrysan — APPROVED** [review]

**Pull request merged**